astropy/units/core.py

```diff
@@ -1968,6 +2027,16 @@ class CompositeUnit(UnitBase):
         A sequence of powers (in parallel with ``bases``) for each
         of the base units.
     """
+    def __new__(cls, scale=None, bases=None, powers=None, decompose=False,
+                decompose_bases=set(), _error_check=True):
+        # Have to set defaults of None since we can enter here from picking.
```

Comment:
I'm not sure I understand the purpose of this comment.  What does setting `None` do for `__init__` args that are positional and not optional, like `scale`, `bases`, and `powers`?  This sounds to me, instead, like an argument for implementing a [`__getnewargs__`](https://docs.python.org/2/library/pickle.html#object.__getnewargs__) for this class.